prime-router/src/test/kotlin/fhirengine/azure/FHIRRouterIntegrationTests.kt

```diff
@@ -0,0 +1,1116 @@
+                    "phd.fhir-elr-no-transform",
+                    fhirSenderWithNoTransform.topic,
+                    "phd.x"
+                )
+            }.map {
+                it.serialize()
+            }
+
+            // filter should permit message and should not mangle message
+            verify(exactly = 1) {
+                QueueAccess.sendMessage(
+                    elrTranslationQueueName,
+                    match {
+                        expectedRouteQueueMessages.contains(it)
+                    }
+                )
+            }
+
+            // make sure action table has a new entry
+            checkActionTable(listOf(TaskAction.convert, TaskAction.receive, TaskAction.route))
+
+            // check that the ACTION_LOG table has the expected but strange entry
+            checkActionLogTable(expectedButStrangeObservationFilterEntry)
+        }
+    }
+
+    @Test
+    fun `should respect jurisdictional filter and not send message`() {
+        // set up
+        val reportContents =
+            listOf(
+                File(VALID_FHIR_URL).readText()
+            ).joinToString()
+
+        val reportPair = createReportsWithLineage(reportContents)
+        val convertReport = reportPair.second
+        val queueMessage = generateQueueMessage(convertReport, reportContents, fhirSenderWithNoTransform)
+        val fhirFunctions = createFHIRFunctionsInstance()
+
+        // make sure action table has only what we put in there
+        checkActionTable(listOf(TaskAction.convert, TaskAction.receive))
+
+        // execute
+        val receiverSetupData = listOf(ReceiverSetupData("x", jurisdictionalFilter = jurisdictionalFilterIl))
+        val receivers = createReceivers(receiverSetupData)
+        val org = createOrganizationWithReceivers(receivers)
+        val fhirRouter = createFHIRRouter(org)
+        fhirFunctions.doRoute(queueMessage, 1, fhirRouter)
+
+        // no messages should have been routed due to filter
+        verify(exactly = 0) {
+            QueueAccess.sendMessage(elrTranslationQueueName, any())
+        }
+
+        // make sure action table has a new entry
+        checkActionTable(listOf(TaskAction.convert, TaskAction.receive, TaskAction.route))
+
+        // we don't log applications of jurisdictional filter to ACTION_LOG at this time
+        ReportStreamTestDatabaseContainer.testDatabaseAccess.transact { txn ->
+            val actionLogRecords = DSL.using(txn)
+                                      .select(Tables.ACTION_LOG.asterisk())
+                                      .from(Tables.ACTION_LOG)
+                                      .fetchInto(ActionLog::class.java)
+
+            assert(actionLogRecords.isEmpty())
+        }
+
+        // we don't log jurisdictional filter actions
+        checkActionLogTable(listOf())
+    }
+
+    @Test
+    fun `should respect full quality filter and not send message`() {
```

Comment:
I like all these tests you added because it illustrates another advantage of simplifying the filters implementation: less tests!